src/lib.rs

```diff
@@ -1,12 +1,1184 @@
 #![allow(non_snake_case)]
-
-#[macro_use] extern crate enum_primitive;
 #[macro_use] extern crate log;
 
 extern crate libusb;
 extern crate byteorder;
-extern crate num;
-extern crate encoding;
 extern crate time;
 
-pub mod ptp;
+use byteorder::{ReadBytesExt, WriteBytesExt, LittleEndian, ByteOrder};
+use std::io::prelude::*;
+use std::io::Cursor;
+use std::io;
+use std::fmt;
+use std::time::Duration;
+
+#[derive(Debug, PartialEq)]
+#[repr(u16)]
+pub enum PtpContainerType {
+    Command = 1,
+    Data = 2,
+    Response = 3,
+    Event = 4,
+}
+
+impl PtpContainerType {
+    fn from_u16(v: u16) -> Option<PtpContainerType> {
+        use self::PtpContainerType::*;
+        match v {
+            1 => Some(Command),
+            2 => Some(Data),
+            3 => Some(Response),
+            4 => Some(Event),
+            _ => None
+        }
+    }
+}
+
+pub type ResponseCode = u16;
+
+#[allow(non_upper_case_globals)]
+pub mod StandardResponseCode {
+    pub type Value = super::ResponseCode;
+    
+    pub const Undefined: Value = 0x2000;
+    pub const Ok: Value = 0x2001;
+    pub const GeneralError: Value = 0x2002;
+    pub const SessionNotOpen: Value = 0x2003;
+    pub const InvalidTransactionId: Value = 0x2004;
+    pub const OperationNotSupported: Value = 0x2005;
+    pub const ParameterNotSupported: Value = 0x2006;
+    pub const IncompleteTransfer: Value = 0x2007;
+    pub const InvalidStorageId: Value = 0x2008;
+    pub const InvalidObjectHandle: Value = 0x2009;
+    pub const DevicePropNotSupported: Value = 0x200A;
+    pub const InvalidObjectFormatCode: Value = 0x200B;
+    pub const StoreFull: Value = 0x200C;
+    pub const ObjectWriteProtected: Value = 0x200D;
+    pub const StoreReadOnly: Value = 0x200E;
+    pub const AccessDenied: Value = 0x200F;
+    pub const NoThumbnailPresent: Value = 0x2010;
+    pub const SelfTestFailed: Value = 0x2011;
+    pub const PartialDeletion: Value = 0x2012;
+    pub const StoreNotAvailable: Value = 0x2013;
+    pub const SpecificationByFormatUnsupported: Value = 0x2014;
+    pub const NoValidObjectInfo: Value = 0x2015;
+    pub const InvalidCodeFormat: Value = 0x2016;
+    pub const UnknownVendorCode: Value = 0x2017;
+    pub const CaptureAlreadyTerminated: Value = 0x2018;
+    pub const DeviceBusy: Value = 0x2019;
+    pub const InvalidParentObject: Value = 0x201A;
+    pub const InvalidDevicePropFormat: Value = 0x201B;
+    pub const InvalidDevicePropValue: Value = 0x201C;
+    pub const InvalidParameter: Value = 0x201D;
+    pub const SessionAlreadyOpen: Value = 0x201E;
+    pub const TransactionCancelled: Value = 0x201F;
+    pub const SpecificationOfDestinationUnsupported: Value = 0x2020;
+    
+    pub fn name(v: Value) -> Option<&'static str> {
```

Comment:
They're type aliases of `u16`, not a new type. Could make this a wrapper struct to give it its own type, I guess...